### PR TITLE
Add instanseg

### PR DESCRIPTION
**Pull request by bethac07**

### CODE REVIEW

**Comment by gnodar01** [comment]
I've added a dockerfile for both non-cuda and cuda. They both work on my intel macbook. Both images have been pushed to dockerhub.

Would be nice to have someone test the cuda image on an actual nvidia machine, to see if GPU mode works.

I did both after all because of the massive size difference. Compressed, the non-cuda is 2.33 GB and the cuda image is 7.97 GB. I have no idea why there is that large of a discrepancy. Yes they use different bases (but still on Ubuntu 22.04), and cuda-compiled versions of various packages, but I wouldn't have expected THAT much of a difference.

`nvidia/cuda:12.3.1-base-ubuntu22.04` is 30.82 MB, and `ubuntu:jammy` is 28.17 MB.

**Comment by bethac07** [comment]
Thanks man, nice work!

I finally also traced the issue I was seeing on my side, it is a "Beth didn't upgrade her computer in a long time and there's a now-patched bug in Rosetta" :shame: :shame: :shame:

**Comment by gnodar01** [comment]
Is it intentional to have the rather large `_build` folder, containing docs here in this PR (or in general not gitignored)?

**Comment by rajavishah** [comment]
Ohh sorry, forgot to insert the folder name in gitignore! - Will fix that in next commit

**Comment by rajavishah** [comment]
Resolves #77, and instanseg algorithm 

@gnodar01 feel free to review

**Review by gnodar01 — CHANGES_REQUESTED** [review]
I think at this point we can get rid of `Dockerfile_conda`, `Dockerfile_mamba`, and `Dockerfile_working_mamba`

File: noxfile.py [hunk 1]
```diff
@@ -22,67 +22,72 @@ def run_generate(session):
...
+        algorithm_folder_name = config.get('algorithm_folder_name', None)
+
+        # Save the algorithm folder name in a file
+        with open('/tmp/algorithm_folder_name.txt', 'w') as file:
+            file.write(algorithm_folder_name)
 
+        # Attempt to pull the image from DockerHub
+        try:
+            print(f'Trying to pull the Docker image: {docker_image_name}')
+            session.run('docker', 'pull', "--platform", "linux/amd64", "bilayer/instanseg:1.0.2")
```
Comment (gnodar01) [inline noxfile.py:55]: Shouldn't be hardcoding "bilayer/instanseg" here.

**Review by gnodar01 — COMMENTED** [review]
Needs to be rebased on main as well, with conflicts.

**Comment by bethac07** [comment]
Closing for now, as it's seeming more likely we want to probably only cherry pick very small parts of this branch out for real, and other parts (like the various dockerfiles) will only be for documentation. Please don't delete the branch though until that work is done.

**Pull request closed without merging**